File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.rb
# @param {Integer[]} nums
# @return {Integer[]}
def smaller_numbers_than_current(nums)
    hash = Hash.new(-1)
    arr = nums.sort
    arr.each_with_index do |num, idx|
        hash[num] = idx if hash[num] == -1
    end
    ret_val = []
    nums.each { |num| ret_val << hash[num] }
    ret_val
end

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        arr = []
        for i in nums:
            counter = 0
            for j in nums:
                if i != j:
                    if j < i:
                        counter += 1
            arr.append(counter)
        return arr  